### PR TITLE
Fix maximum concurrency for jenkins-operator

### PR DESCRIPTION
**Pull request by kargakis**
Tackling the controllers one by one because the refactoring is quite big.

Part of https://github.com/kubernetes/test-infra/issues/4109

/assign @nlandolfi @spxtr 
/area prow
/kind bug

Signed-off-by: Michail Kargakis <mkargaki@redhat.com>

### CODE REVIEW

**Comment by kargakis** [comment]
Debugging this today - first run, max_concurrency doesn't seem to be respected.

**Comment by kargakis** [comment]
> Debugging this today - first run, max_concurrency doesn't seem to be respected.

Figured this out - I wasn't updating the pendingJobs map in places I should. Works like a charm now!

File: prow/jenkins/controller.go [hunk 1]
```diff
@@ -90,30 +104,7 @@ func (c *Controller) incrementNumPendingJobs(job string) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
```
Comment (BenTheElder) [inline prow/jenkins/controller.go]: Nit: we should probably make the blank line here consistent between getNumPendingJobs and incrementNumPendingJobs (remove it?)

**Comment by BenTheElder** [comment]
I think this is LGTM, but I'd like to get @spxtr to look at it.

File: prow/jenkins/controller.go (hunk 1 above)
Comment (kargakis) [reply prow/jenkins/controller.go]: Yeah, I just did it w/o reading this comment :)

**Comment by spxtr** [comment]
> Tackling the controllers one by one because the refactoring is quite big.

This worries me. Do you think it's worth sharing more code between the controllers such that this isn't a problem?

/lgtm

**Comment by kargakis** [comment]
> This worries me. Do you think it's worth sharing more code between the controllers such that this isn't a problem?

I was planning on moving more common pieces in a shared package but there are parts that cannot be dryed and that's reasonable I think.

**Pull request merged**